astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -362,6 +364,81 @@ def test_astype(self):
             int32q = self.q.astype("int32")
             assert_array_equal(np.astype(int32q, "int32"), int32q)
 
+    @needs_array_function
+    @pytest.mark.parametrize("angle_cls", [u.Quantity, Angle])
```

Comment:
I think this is a bit of overkill. I suggest removing the parametrization, and just add a single test function that does a subclass. E.g., 
```
test_arange_subclass(self):
    class AngularUnits(u.SpecificTypeQuantity):
        _equivalent_unit = u.radian

    ...
```
That also means we do not have to import `Angle`.